csrc/index_compute.cpp

```diff
@@ -3171,4 +3171,83 @@ Val* Index::eye(
   return GpuLower::current()->commonScalarMap().hoistScalar(result, loops);
 }
 
+Val* Index::cpAsyncBulkIndex(
+    TensorView* tv,
+    const std::vector<kir::ForLoop*>& loops) {
+  using namespace tma;
+
+  TORCH_INTERNAL_ASSERT(
+      tv->getMemoryType() == MemoryType::Global,
+      "cpAsyncBulkIndex is only for global memory tensors");
+  TORCH_INTERNAL_ASSERT(
+      tv->getMaybeRFactorDomain() == tv->getLeafDomain(), "not supported yet");
+  TORCH_INTERNAL_ASSERT(
+      tv->getMaybeAllocationDomain() == tv->getLeafDomain(),
+      "not supported yet");
+  for (auto id : tv->getMaybeRFactorDomain()) {
+    TORCH_INTERNAL_ASSERT(
+        id->isBulk(),
+        "cpAsyncBulkIndex only support whole tensor copy for now.");
+  }
+
+  int64_t dim = (int64_t)tv->nDims();
+  TORCH_INTERNAL_ASSERT(dim > 0);
+  int64_t itemsize = dataTypeSize(tv->dtype());
+
+  auto metadata = IrBuilder::metadataExpr(tv);
+  auto global_address = IrBuilder::getAttrExpr(metadata, "data");
+  auto global_dim =
+      // Reverse array to convert from row major to column major
```

Comment:
can you add a comment about why need to convert to column major? didn't find this info from the doc :(

Reply:
Added. This is just a requirement of hardware, and can be inferred from this part in the doc:
```C++
‎    globalStrides[0] = globalDim[0] * elementSizeInBytes(tensorDataType) + padding[0];
          for (i = 1; i < tensorRank - 1; i++)
              globalStrides[i] = globalStrides[i – 1] * (globalDim[i] + padding[i]);
              assert(globalStrides[i] >= globalDim[i]);
```